### DISCUSSION TITLE
change exe details like company name, icon, etc

### DISCUSSION THREAD

**Discussion by si458** [Ylianst/MeshCentral#4051] [Ideas]
started new discussion to seperate ideas etc

it would be amazing to be able to change the meshagent exe icon and also the company name of the exe

i understand this can be done with the likes of resource hacker or electrons rcedit but they arent pure nodejs sadly

https://github.com/electron/rcedit (c++ exe i believe) - https://github.com/electron/node-rcedit (node package to run rcedit)

https://github.com/si458/changeexe (uses resource hacker sadly)

**Comment by Ylianst** [comment]
Ha yes, as I mentioned in #4049, this is exactly the plan. The first step was to get MeshCentral to be able to code sign the agent. With the new [Authenticode-JS](https://www.npmjs.com/package/authenticode-js) that we released and upcoming integration into MeshCentral the agent will be signed by each server automatically. Once the agent is signed by the server, this opens up the opportunity now to modify and customize the agent without worrying about breaking the signature.

If there was a purely NodeJS way to edit .exe resources, I would get this added feature added in a day, but since I can't find a good solution I am going to have to build it. The 3 links above all use Windows OS calls to make the changes, so they will not work for us.
